index.js

```diff
@@ -70,7 +95,41 @@ class Main {
     metrics.currentBlock.set(this.worker.lastConfirmedBlock);
     metrics.requestsCounter.inc(this.worker.getNewRequestsCount());
     metrics.requestsResponseTime.observe(new Date() - this.worker.lastRequestStartTime);
-    metrics.lastExportedBlock.set(this.worker.lastExportedBlock);
+    if (BLOCKCHAIN === 'eth') {
+      metrics.lastExportedBlock.set(this.lastProcessedPosition.blockNumber);
+    } else {
+      metrics.lastExportedBlock.set(this.worker.lastExportedBlock);
+    }
+  }
+
+  /**
+   * Helper method for checking the blockchain in order to use the correct primary key functionality.
+   * @param {Array} buffer The array of events that would be accordingly updated with the primary keys.
+   */
+  applyPrimaryKeyLogic(buffer) {
+    if (BLOCKCHAIN === 'eth') {
```

Comment:
You can leverage polymorphism and have 'decorateWithPrimaryKeys' method in different workers, instead of moving blockchain specific logic in the `index.js`.